src/components/Map.js

```diff
@@ -1,229 +1,196 @@
-/* Main map component based on leaflet map.
- *
- *
- */
-import React, {Component} from 'react'
+import React from 'react'
 import {RENDERING_CLUSTERS, RENDERING_HEATMAP, RENDERING_MARKERS, RENDERING_POLYLINE} from "./Layer";
 import L from 'leaflet';
 import 'leaflet.heat';
 import 'leaflet.markercluster';
 import 'leaflet.markercluster/dist/MarkerCluster.css';
 import 'leaflet.markercluster/dist/MarkerCluster.Default.css';
 
+/* 
+ * Main map component based on leaflet map.
+ */
+export const Map = React.memo(({layers}) => {
+
+	const mapRef = React.useRef();
+	const mapElementRef = React.createRef();
+
+	const mapOverlaysRef = React.useRef({});
+	const layerControlRef = React.useRef(new L.control.layers([]));
+
+	React.useEffect(() => {
```

Comment:
Do we expect the map and base layers to change after first rendering? Or maybe I am missing the purpose of useEffect, have to checkout your branch and test locally :)

Reply:
This replaces the `componentDidMount` lifecycle and is only run once